src/main/java/org/sonarlint/intellij/core/server/events/BindingChangeListener.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2021 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.core.server.events
+
+import com.intellij.openapi.project.Project
+import org.sonarlint.intellij.common.util.SonarLintUtils.getService
+import org.sonarlint.intellij.core.ProjectBinding
+import org.sonarlint.intellij.messages.ProjectBindingListener
+
+class BindingChangeListener(project: Project) : ProjectBindingListener {
+    override fun bindingChanged(previousBinding: ProjectBinding, newBinding: ProjectBinding) {
+        getService(ServerEventsService::class.java).autoSubscribe(previousBinding)
```

Comment:
What is it for? Why subscribing to previous binding?

Reply:
This naming of `ProjectBinding` and `previousBinding `is really misleading, I plan to change it

This `bindingChanged` callback can be called when you switch from one connection to another, so you need to re-register for the previous connection with up-to-date project keys